### PR TITLE
Add verifiable presentation method

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
Add a method to `wallet-connectors` for requesting verifiable presentations from wallets.

## Changes
- Added a new method to the `WalletConnection` interface for requesting verifiable presentations.
- Implemented the method for browser wallet and WalletConnect.
- Added a sample project that utilizes the added method.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: samples/proofs/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                <Col sm={3}>
+                    <Button variant="primary" onClick={handleSubmit} disabled={!connection || isWaiting}>
+                        {isWaiting ? 'Waiting for proof...' : 'Request proof'}
+                    </Button>
+                </Col>
+            </Form.Group>
+            <Row>
+                {error && <Alert variant="danger">{error}</Alert>}
+                {verifiablePresentation && (
+                    <Accordion defaultActiveKey="0">
```
Comment (bisgardo) [inline samples/proofs/src/App.tsx:108]: There isn't much else on the page so maybe accordion is overkill?

File: packages/wallet-connectors/CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## [0.5.0]
```
Comment (bisgardo) [inline packages/wallet-connectors/CHANGELOG.md:8]:
```suggestion
## [Unreleased]

## [0.5.0]
```

But really I generally prefer setting version numbers (which in this file should include release date) in a separate "release" PR.

**Review by bisgardo — CHANGES_REQUESTED** [review]
`@concordium/wallet-connectors` dependency from react-components needs to be bumped.

**Comment by bisgardo** [comment]
`requestVerifiablePresentation` is an odd name to me. Is this standard terminology? Otherwise I think I have to go though a little too imaginative guesswork to figure out what it's doing.

**Comment by abizjak** [comment]
> `requestVerifiablePresentation` is an odd name to me. Is this standard terminology? Otherwise I think I have to go though a little too imaginative guesswork to figure out what it's doing.
> 
> Same with just the "Verifiable Presentation" part.

Verifiable presentation is the term for the thing that is being requested.

File: samples/proofs/src/App.tsx (hunk 1 above)
Comment (orhoj) [reply samples/proofs/src/App.tsx:108]: It is much nicer than just using `<pre>`. I don't know react-bootstrap well enough to know if there are better alternatives. I think this looks nice for a sample.

Feel free to suggest an alternative from that library for displaying a big JSON blob.

File: packages/wallet-connectors/CHANGELOG.md (hunk 2 above)
Comment (bisgardo) [reply packages/wallet-connectors/CHANGELOG.md]: The suggestion was to include `## [Unreleased]`, not replace `## [0.5.0]`. But this is also fine :)

**Review by bisgardo — APPROVED** [review]

**Pull request merged**